beetsplug/lastgenre/__init__.py

```diff
@@ -410,7 +421,7 @@ def lastgenre_func(lib, opts, args):
                     self._log.info(
                         "genre for album {0} ({1}): {0.genre}", album, src
                     )
-                    album.store()
+                    album.store(inherit=False)
```

Comment:
Perhaps this ought to be configurable? I suppose I'd expect, in the "normal" (default) use of the lastgenre plugin, people _do_ want "inheritance," i.e., they want all the tracks on their albums to have the same genre.

Reply:
Yes you are absolutely right, that is an oversight. I was so focused on getting it to behave as expected when `source: track` is configured that I didn't think of that.

Let me explain with an example:

```
$ beet ls -a -f '$albumartist - $album - $title |$genre' behind the front
Black Eyed Peas - Behind the Front - $title |

$ beet ls -f '$albumartist - $album - $title |$genre' behind the front
Black Eyed Peas - Behind the Front - Karma |
Black Eyed Peas - Behind the Front - Be Free |
Black Eyed Peas - Behind the Front - Communication |
Black Eyed Peas - Behind the Front - What It Is |
```

So if I now change the code to `album.store(inherit=True)`

and having configured `source: track`, I would expect to get an album genre applied to my album only, but also fetch individual genres for each track.

The pitty is that now I do not get my track genres since they have already inherited the album genre, the "keep allowed" magic kicks in and will make sure that all genres are kept. So in the end all my tracks get the album's genre and `source: track` is not doing as expected:

```
$ beet -vv lastgenre -a behind the front
user configuration: /Users/jojo/.config/devbeets/config.yaml
data directory: /Users/jojo/.config/devbeets
plugin paths:
Sending event: pluginload
library database: /Users/jojo/.config/devbeets/library.db
library directory: /Users/jojo/Music/dev-beets
Sending event: library_opened
Parsed query: AndQuery([AnyFieldQuery('behind', ('album', 'albumartist', 'genre'), SubstringQuery), AnyFieldQuery('the', ('album', 'albumartist', 'genre'), SubstringQuery), AnyFieldQuery('front', ('album', 'albumartist', 'genre'), SubstringQuery)])
Parsed sort: NullSort()
lastgenre: genre for album Black Eyed Peas - Behind the Front |Rap, Funk, Hip Hop (album): Rap, Funk, Hip Hop
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
Sending event: database_change
lastgenre: genre for track Black Eyed Peas - Behind the Front - Karma |Funk, Rap, Hip Hop (keep): Funk, Rap, Hip Hop
Sending event: write
Sending event: after_write
Sending event: database_change
lastgenre: genre for track Black Eyed Peas - Behind the Front - Be Free |Funk, Rap, Hip Hop (keep): Funk, Rap, Hip Hop
Sending event: write
Sending event: after_write
Sending event: database_change
lastgenre: genre for track Black Eyed Peas - Behind the Front - Communication |Funk, Rap, Hip Hop (keep): Funk, Rap, Hip Hop
Sending event: write
Sending event: after_write
Sending event: database_change
lastgenre: genre for track Black Eyed Peas - Behind the Front - What It Is |Funk, Rap, Hip Hop (keep): Funk, Rap, Hip Hop
Sending event: write
Sending event: after_write
Sending event: cli_exit
```


I'm sure there is an easy way out of this...but using `inherit=False` only was not it!

Reply:
So the only way now to get out of this I could try to set tracks genres by simply matching my tracks only and not the album:

```
beet lastgenre -A behind the front
```

that will again just fall back to do the "keep allowed magic" and my track genre's sill still be those of the album's

So the only way out of this is to use force again on my tracks only:

```
$ beet lastgenre -f -A behind the front
```

and these "extra miles" were not was I was expecting when having enabled "track agnostic" genre fetching by setting `source: track`

Does that make sense? :-)

Reply:
And for reference, the actual meaning of setting sources to `album`, `artist`, or track:
https://github.com/beetbox/beets/blob/1fdf075402415fd0b58776a7de760a161b684a60/beetsplug/lastgenre/__init__.py#L154-L165

Reply:
Oh, that was actually easy to fix. See 321d87e2

Reply:
Yes, that seems about right to me!!